packages/main/src/MenuItem.ts

```diff
@@ -534,16 +541,28 @@ class MenuItem extends ListItem implements IMenuItem {
 		this._closeOtherSubMenus(item);
 	}
 
+	_isSpace(e: KeyboardEvent) {
+		this._shiftPressed = this._isCheckable && isSpaceShift(e);
+		return isSpace(e) || this._shiftPressed || (!this._isCheckable && isSpaceShift(e));
+	}
+
+	_isEnter(e: KeyboardEvent) {
+		this._shiftPressed = this._isCheckable && isEnterShift(e);
+		return isEnter(e) || this._shiftPressed || (!this._isCheckable && isEnterShift(e));
+	}
+
+	_onclick(e: MouseEvent) {
+		this._shiftPressed = this._isCheckable && e.shiftKey;
+		super._onclick(e);
```

Comment:
Keyup handler should be added in order to reset shiftPressed variable once the shift key is released.